src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/PropertySymbolReferenceFinder.cs

```diff
@@ -172,51 +170,32 @@ private static bool IsForEachProperty(IPropertySymbol symbol)
                         continue;
                     }
 
-                    var location = argumentList.SyntaxTree.GetLocation(new TextSpan(argumentList.SpanStart, 0));
-
-                    var symbolUsageInfo = GetSymbolUsageInfo(node, semanticModel, syntaxFacts, semanticFacts, cancellationToken);
-                    locations.Add(new FinderLocation(
-                        node, new ReferenceLocation(document, null, location, isImplicit: false, symbolUsageInfo, GetAdditionalFindUsagesProperties(node, semanticModel, syntaxFacts), candidateReason: reason)));
+                    (matched, reason) = symbolsMatch(node, semanticModel);
                 }
-            }
-
-            return locations.ToImmutableAndFree();
-        }
-
-        private async Task<ImmutableArray<FinderLocation>> FindIndexerCrefReferencesAsync(
-            IPropertySymbol symbol, Document document, SemanticModel semanticModel,
-            FindReferencesSearchOptions options, CancellationToken cancellationToken)
-        {
-            if (options.AssociatePropertyReferencesWithSpecificAccessor)
-            {
-                // can't find indexer get/set accessors in a cref.
-                return ImmutableArray<FinderLocation>.Empty;
-            }
-
-            var symbolsMatch = GetStandardSymbolsNodeMatchFunction(symbol, document.Project.Solution, cancellationToken);
-
-            var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
-            var semanticFacts = document.GetLanguageService<ISemanticFactsService>();
-
-            // Now that we have Doc Comments in place, We are searching for References in the Trivia as well by setting descendIntoTrivia: true
-            var indexerMemberCrefs = syntaxRoot.DescendantNodes(descendIntoTrivia: true)
-                                               .Where(syntaxFacts.IsIndexerMemberCRef);
+                else if (syntaxFacts.IsConditionalAccessExpression(node))
+                {
+                    // For a ConditionalAccessExpression the whenNotNull component is the indexer reference we are looking for
+                    syntaxFacts.GetPartsOfConditionalAccessExpression(node, out var expression, out indexerReference);
 
-            var locations = ArrayBuilder<FinderLocation>.GetInstance();
+                    (matched, reason) = symbolsMatch(indexerReference, semanticModel);
+                }
+                else
+                {
+                    // For an IndexerMemberCRef the node itself is the indexer we are looking for.
+                    indexerReference = node;
 
-            foreach (var node in indexerMemberCrefs)
-            {
-                cancellationToken.ThrowIfCancellationRequested();
+                    (matched, reason) = symbolsMatch(node, semanticModel);
+                }
```

Comment:
consider extracting this whole `if` to a helper method or `static local function`.